app/services/course/assessment/programming_evaluation_service.rb

```diff
@@ -0,0 +1,95 @@
+# Sets up a programming evaluation, queues it for execution by evaluators, then returns the results.
+class Course::Assessment::ProgrammingEvaluationService
+  # The default timeout for the job to finish.
+  DEFAULT_TIMEOUT = Course::Assessment::ProgrammingEvaluation::TIMEOUT
+
+  # Represents a result of evaluating a package.
+  class Result
+    attr_reader :stdout, :stderr, :test_report
+
+    def initialize(stdout, stderr, test_report)
+      @stdout = stdout
+      @stderr = stderr
+      @test_report = test_report
+    end
+  end
+
+  class << self
+    # Executes the provided package.
+    #
+    # @param [Course] course The course which this evaluation belongs to. This is necessary to
+    #   determine which workers get access to execute the package.
+    # @param [Polyglot::Language] language The language runtime to use to run this package.
+    # @param [Fixnum] memory_limit The memory limit for the evaluation, in MiB.
+    # @param [Fixnum|ActiveSupport::Duration] time_limit The time limit for the evaluation, in
+    #   seconds.
+    # @param [String] package The path to the package.
+    # @param [nil|Fixnum] timeout The duration to elapse before timing out. When the operation
+    #   times out, a +Timeout::TimeoutError+ is raised. This is different from the time limit in
+    #   that the time limit affects only the run time of the evaluation. The timeout includes
+    #   waiting for abn evaluator, setting up the environment etc.
+    # @return [Result] The result of evaluating the template.
+    #
+    # @raise [Timeout::Error] When the operation times out.
+    def execute(course, language, memory_limit, time_limit, # rubocop:disable Metrics/ParameterLists
```

Comment:
it's passed from the database record... ostensibly I could use hash arguments, but I think this is clearer. Or do you think hash arguments are better?